blog-content/using-github-as-a-headless-cms.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Using Github as a Headless CMS 
+createdAt: 2023-06-01T12:00:00Z
+readingTime: 12
+authorFirstName: Abhishek
+authorLastName: More
+authorTitle: Software Engineer
+authorTwitter: ''
+authorLinkedIn: 'https://www.linkedin.com/in/abhishek-more-linked/'
+authorGithub: 'https://github.com/Abhishek-More'
+authorWebsite: 'https://abhishekmore.com'
+authorPFP: 'https://tamuhack.org/static/th-2022/headshots/webp/abhishek.webp'
+tags: Highlight Engineering
+metaTitle: Using Github as a Headless CMS 
+---
+
+A content-management system (CMS) is software used to create, manage, and host content for blogs, docs, etc.
+It is crucial to use a CMS for any frequently updating content on your site, as it allows for:
+
+- Easy content creation and editing
+- Centralized content management
+- Collaboration and workflow management
+
+Recently, Highlight switched over from [Hygraph](https://hygraph.com) to GitHub for managing blog content, and we've been loving it!
+
+## Why GitHub is the best solution for us
+
+### Speedy Build Times 💨
+
+At the time of writing, we have 48 blog posts. At build time, post content is fetched from the CMS and each post page is statically generated
+on the server (See [Static Site Generation](https://nextjs.org/docs/pages/building-your-application/rendering/static-site-generation)). 
+This means that we need to make 48 separate API calls to Hygraph to retrieve post content. However, due to API rate limits, we had to introduce a delay for each request. While this worked fine initially, our build time started to suffer as the number of posts increased.
+
+With GitHub, all of our posts are written in markdown files and stored in the repository itself. The server is now able to read the file system and instantly fetch post content without having to make a single API call.
+
+![build-time-over-time](/images/blog/using-github-as-headless-cms/blog-build-time.png)
+
+Our build time for blog content is now down to **~2 seconds** and will remain constant as the number of posts increases. 
+
+### Improved Developer Experience 💡
+
+For our technical content writers, GitHub couldn't be better! With the limited team members allowed by Hygraph, not everyone was able to review blog posts in Hygraph before they get published. Now, our engineers can make PRs and let everyone see the hosted blog post formatted exactly how its supposed to be.
+
+Additionally, we get access to GitHub features, including Issues, Actions, and Projects. This ensures better organization and also provides a centralized platform for collaboration and feedback. 
+
+[Check out the PR for this one!](https://github.com/highlight/highlight/pull/5635)
+
+## How we did it 
```

Comment:
Can you put this before the section about what the benefits are? Imo, this is more interesting and a better hook for a technical audience.